lib/trino-orc/src/main/java/io/trino/orc/reader/SliceDictionaryColumnReader.java

```diff
@@ -185,7 +185,7 @@ private Block readNullBlock(boolean[] isNull, int nonNullCount)
         if (nonNullValueTemp.length < minNonNullValueSize) {
             nonNullValueTemp = new int[minNonNullValueSize];
             nonNullPositionList = new int[minNonNullValueSize];
-            systemMemoryContext.setBytes(sizeOf(nonNullValueTemp) + sizeOf(nonNullPositionList));
+            updateMemory();
```

Comment:
Please update the commit message to something like `Account for memory usage of dictionary in SliceDictionaryColumnReader` and add any relevant info about the fixed problem in the commit description.

Reply:
Commit message has been updated.